### PR TITLE
Feature/mfmehari/krok obio

### PR DESCRIPTION
**Pull request by mfmehari**
OBIO (NOBM) is integrated into the Krok Model. 
It is still in testing phase.
It is a zero diff PR.

### CODE REVIEW

**Comment by sanAkel** [comment]
Thanks @mfmehari I'll take a look at this tomorrow (late in the day).

Can you list all the ways you tested this PR. Example coupled ( which ocean, what resolution(s)), also you must do an uncoupled model run - it should preserve answers in both (uncoupled and coupled) modes- this is needed because you have changed source code at a lot of places.

**Review by tclune — APPROVED** [review]
cmake changes ok

**Comment by mfmehari** [comment]
> Thanks @mfmehari I'll take a look at this tomorrow (late in the day).
> 
> Can you list all the ways you tested this PR. Example coupled ( which ocean, what resolution(s)), also you must do an uncoupled model run - it should preserve answers in both (uncoupled and coupled) modes- this is needed because you have changed source code at a lot of places.

@sanAkel 
It is tested with the coupled model run:
AGCM: C12 (IM:12 x JM:72) and LM:72
OGCM:  IM:72 x JM:36 and LM:50
Okay, I will do the uncoupled run.
Thank you!

**Comment by mfmehari** [comment]
> > Thanks @mfmehari I'll take a look at this tomorrow (late in the day).
> > Can you list all the ways you tested this PR. Example coupled ( which ocean, what resolution(s)), also you must do an uncoupled model run - it should preserve answers in both (uncoupled and coupled) modes- this is needed because you have changed source code at a lot of places.
> 
> @sanAkel It is tested with the coupled model run: AGCM: C12 (IM:12 x JM:72) and LM:72 OGCM (MOM6): IM:72 x JM:36 and LM:50 Okay, I will do the uncoupled run. Thank you!

**Comment by mfmehari** [comment]
> > Thanks @mfmehari I'll take a look at this tomorrow (late in the day).
> > Can you list all the ways you tested this PR. Example coupled ( which ocean, what resolution(s)), also you must do an uncoupled model run - it should preserve answers in both (uncoupled and coupled) modes- this is needed because you have changed source code at a lot of places.
> 
> @sanAkel It is tested with the coupled model run: AGCM: C12 (IM:12 x JM:72) and LM:72 OGCM (MOM6): IM:72 x JM:36 and LM:50 Okay, I will do the uncoupled run. Thank you!

@sanAkel Hi Santha, 
Tested the uncoupled run and it has a zero diff.
AGCM: C12 (im:12 x JM:72) and LM:72
Data_Ocean Horizontal Resolution code: o1 (1  -deg,  360x180  Reynolds)
Thank you!

**Comment by sanAkel** [comment]
@mfmehari Please update when you have:

1. Added and,
2. Tested a `2d mask` export from MOM5 _plug_. 

Because ⬆️ needs to happen in [Ocean GC repository](https://github.com/GEOS-ESM/GEOS_OceanGridComp), adding a contingency.

Once you have above 1 and 2 done, please remove this contingency. Thanks!

**Review by sdrabenh — APPROVED** [review]

**Review by bena-nasa — APPROVED** [review]

**Review by sdrabenh — APPROVED** [review]

**Review by gmao-rreichle — APPROVED** [review]
Changes ok from Land perspective.  Would appreciate a more descriptive PR title.

File: GEOS_GcmGridComp.F90 [hunk 1]
```diff
@@ -206,7 +206,7 @@ subroutine SetServices ( GC, RC )
     endif
 
     call MAPL_GetResource ( MAPL, DO_OBIO, Label="USE_OCEANOBIOGEOCHEM:", DEFAULT=0, _RC)
-    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN, Label="USE_DATAATM:", DEFAULT=0, _RC)
+    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN,  Label="USE_DATA_ATM4OCN:", DEFAULT=0, _RC)
```
Comment (sanAkel) [inline GEOS_GcmGridComp.F90:209]: Logical (`True` or `False`) are preferred than int for `DO_DATA_ATM4OCN`

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOS_SurfaceGridComp.F90 [hunk 2]
```diff
@@ -233,7 +234,7 @@ subroutine SetServices ( GC, RC )
 
     ! Are we running DataAtm?
     !------------------------
-    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN, Label="USE_DATAATM:" , DEFAULT=0, RC=STATUS)
+    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN, Label="USE_DATA_ATM4OCN:" , DEFAULT=0, RC=STATUS)
```
Comment (sanAkel) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOS_SurfaceGridComp.F90:237]: See ⬆️ comment reg preference for logicals

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp/GEOS_SaltWaterGridComp.F90 [hunk 3]
```diff
@@ -130,6 +127,10 @@ subroutine SetServices ( GC, RC )
 
     call MAPL_GetResource ( MAPL, DO_OBIO,            Label="USE_OCEANOBIOGEOCHEM:", DEFAULT=0, _RC)
 
+! Are we running DataAtm?
+!------------------------
+    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN,  Label="USE_DATA_ATM4OCN:" ,   DEFAULT=0, _RC)
```
Comment (sanAkel) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp/GEOS_SaltWaterGridComp.F90:132]: Logical `DO_DATA_ATM4OCN` is preferred

File: GEOSogcm_GridComp/GEOS_OgcmGridComp.F90 [hunk 4]
```diff
@@ -194,7 +184,7 @@ subroutine SetServices ( GC, RC )
     VERIFY_(STATUS)
     call MAPL_GetResource ( MAPL, DO_OBIO,        Label="USE_OCEANOBIOGEOCHEM:",DEFAULT=0, RC=STATUS)
     VERIFY_(STATUS)
-    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN,     Label="USE_DATAATM:" ,        DEFAULT=0, RC=STATUS)
+    call MAPL_GetResource ( MAPL, DO_DATA_ATM4OCN, Label="USE_DATA_ATM4OCN:" ,  DEFAULT=0,     __RC__ )
```
Comment (sanAkel) [inline GEOSogcm_GridComp/GEOS_OgcmGridComp.F90:187]: Across the board logicals please

**Review by sanAkel — CHANGES_REQUESTED** [review]
@mfmehari Please consider making suggested changes.

**Review by lcandre2 — APPROVED** [review]

**Comment by mathomp4** [comment]
> @mfmehari Please consider making suggested changes.

@sanAkel I guess the only issue is that the rest of the `DO_` resources in the gridcomp around there do not use logicals. 
```
    call MAPL_GetResource ( MAPL, DO_OBIO, Label="USE_OCEANOBIOGEOCHEM:", DEFAULT=0, _RC)
    call MAPL_GetResource ( MAPL, DO_DATASEA, Label="USE_DATASEA:", DEFAULT=1, _RC)
    call MAPL_GetResource ( MAPL, DO_WAVES, Label="USE_WAVES:", DEFAULT=0, _RC)
    call MAPL_GetResource ( MAPL, DO_SEA_SPRAY, Label="USE_SEA_SPRAY:", DEFAULT=0, _RC)
```

I mean, it's what they *should* do but... I guess the question is should we be consistent?

**Comment by sanAkel** [comment]
> > @mfmehari Please consider making suggested changes.
> 
> @sanAkel I guess the only issue is that the rest of the `DO_` resources in the gridcomp around there do not use logicals.
> 
> ```
>     call MAPL_GetResource ( MAPL, DO_OBIO, Label="USE_OCEANOBIOGEOCHEM:", DEFAULT=0, _RC)
>     call MAPL_GetResource ( MAPL, DO_DATASEA, Label="USE_DATASEA:", DEFAULT=1, _RC)
>     call MAPL_GetResource ( MAPL, DO_WAVES, Label="USE_WAVES:", DEFAULT=0, _RC)
>     call MAPL_GetResource ( MAPL, DO_SEA_SPRAY, Label="USE_SEA_SPRAY:", DEFAULT=0, _RC)
> ```
> 
> I mean, it's what they _should_ do but... I guess the question is should we be consistent?

@mathomp4 (and @mfmehari), your ⬆️ question is a relevant one. Following are my _current_ thoughts.
1. [This PR](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/871) is to a repo that has decades of (...) built in. There is absolutely no way we can accomplish `should we be consistent?` **unless** one of _us_ spends _some time_ to _clean_ this (logical) aspect and indeed so many, many, ... others; ultimately we'll be looking at a rewrite! (Just look at surface GC .F90, look at its length and scope? Anyway let's leave it out _here_).
2. I suggestions to @mfmehari are to the point of `what they _should_ do` so things are _better_ going forward. I had the same comments earlier for the _waves_PR_ from @adarmenov and others before it; I'm being consistent in my suggestions.

@mathomp4 did I answer your question(s)?

**Comment by yvikhlya** [comment]
I guess the issue with boolean options is that *.rc format does not support them and such options are read as integers, 0 or 1, and used as integers all over GEOSgcm.

**Comment by mfmehari** [comment]
@sanAkel If we are okay with DO_DATA_ATM4OCN only being in logical form, I could go ahead change its type, thank you!

**Comment by sanAkel** [comment]
> @sanAkel If we are okay with DO_DATA_ATM4OCN only being in logical form, I could go ahead change its type, thank you!

👍 Let's just try to do _good_ here on ...

File: GEOSogcm_GridComp/GEOS_OgcmGridComp.F90 (hunk 4 above)
Comment (mfmehari) [reply GEOSogcm_GridComp/GEOS_OgcmGridComp.F90]: > Across the board logicals please

@sanAkel The type of DO_DATA_ATM4OCN has been changed from int to logical.

**Comment by mfmehari** [comment]
@sanAkel The type of DO_DATA_ATM4OCN has been changed from int to logical.

**Review by sanAkel — APPROVED** [review]

**Review by sdrabenh — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Comment by mfmehari** [comment]
Thank you all! @sdrabenh @mathomp4 @sanAkel @tclune @bena-nasa @lcandre2 @gmao-rreichle

**Pull request merged**